internal/core/thirdparty/tantivy/tantivy-binding/src/array.rs

```diff
@@ -20,10 +30,132 @@ impl RustArray {
     }
 }
 
+impl std::default::Default for RustArray {
+    fn default() -> Self {
+        RustArray {
+            array: std::ptr::null_mut(),
+            len: 0,
+            cap: 0,
+        }
+    }
+}
+
+impl From<Vec<u32>> for RustArray {
+    fn from(vec: Vec<u32>) -> Self {
+        RustArray::from_vec(vec)
+    }
+}
+
 #[no_mangle]
 pub extern "C" fn free_rust_array(array: RustArray) {
     let RustArray { array, len, cap } = array;
     unsafe {
         Vec::from_raw_parts(array, len, cap);
     }
 }
+
+#[repr(C)]
```

Comment:
does the `ffi` type satisfy the rfc https://github.com/rust-lang/rfcs/blob/master/text/2195-really-tagged-unions.md?
Please add some unittest to make sure that it's ok at current compiler.

Reply:
> does the `ffi` type satisfy the rfc https://github.com/rust-lang/rfcs/blob/master/text/2195-really-tagged-unions.md? Please add some unittest to make sure that it's ok at current compiler.

as cbindgen doc says `cbindgen also supports using repr(C)/repr(u8) on non-C-like enums (enums with fields). This gives a C-compatible tagged union layout, as [defined by this RFC 2195](https://github.com/rust-lang/rfcs/blob/master/text/2195-really-tagged-unions.md). repr(C) will give a simpler layout that is perhaps more intuitive, while repr(u8) will produce a more compact layout.`

this enum should be supported. But I will also add a ut to verify it.